src/wp-includes/comment-template.php

```diff
@@ -218,14 +218,50 @@ function get_comment_author_email_link( $linktext = '', $before = '', $after = '
  * @return string The comment author name or HTML link for author's URL.
  */
 function get_comment_author_link( $comment_ID = 0 ) {
+
 	$comment = get_comment( $comment_ID );
 	$url     = get_comment_author_url( $comment );
 	$author  = get_comment_author( $comment );
 
 	if ( empty( $url ) || 'http://' === $url ) {
 		$return = $author;
 	} else {
-		$return = "<a href='$url' rel='external nofollow ugc' class='url'>$author</a>";
+
+		$rel_parts = array();
+		if ( ! wp_is_internal_link( $url ) ) {
+			$rel_parts = array( 'external', 'nofollow' );
+		}
+
+		/**
+		 * Filters the list of non-ugc comment authors.
+		 *
+		 * When displaying the comment author link for any commenters in this list, the "ugc" rel attribute
+		 * will not be added to the anchor's rel attribute.
+		 *
+		 * @since [version]
+		 *
+		 * @param string[] $comment_authors An array of comment author usernames.
+		 */
+		$non_ugc_comment_authors = apply_filters( 'non_ugc_comment_authors', array() );
```

Comment:
Yes... but. Semantically I believe this would be the incorrect place to use a filter/function that determines if the *comment* itself is UGC. This function is retrieving the comment author's link (as submitted by the comment author). So in this particular area we want to know if the *author's link* is UGC, not whether or not the whole comment is UGC.

So... Merging this with your feedback I'd update it to

`comment_author_link_is_ugc`, it'll return a boolean instead of an array of usernames, and as a second param it'll receive the `WP_Comment` object.

The default value will be `true` and if the filtered value of the var is `true` we'll add `ugc` to the `$rel_parts` array.
What do you think about that?

Reply:
I wrote this 3 hours ago and never posted. I rewrote the code and just finished writing tests and now, looking at the new code. I think this particular filter is just unnecessary.

> The function get_comment_author_link() shouldn't modify the author link indiscriminately. Although its output can be filtered, it would be better to prevent link tagging for specific authors. This can be achieved by adding the following code on line 223 of wp-includes/comment-template.php: 

- https://core.trac.wordpress.org/ticket/53290

At the writing of this comment the only available filter is on the full HTML of the anchor via `get_comment_author_link`

With this PR (and updates from feedback) there will be a new filter specifically on the rel attributes as an array before being composed into the HTML anchor.

If a developer desires to use `comment_author_link_rel` to make a specific user automatically not ugc, it can be done so there quite easily. I don't see the need to add two filters here, one to build an allowlist of sorts and another to modify the rel attributes directly.

Incoming PR update applies UGC automatically and can be removed via `comment_author_link_rel`.